File: Assets/Scripts/FollowMouse.js
var depth = 10.0;

function Start ()
{
     Cursor.visible = false;
}

function Update ()

{

     var mousePos = Input.mousePosition;

     var wantedPos = Camera.main.ScreenToWorldPoint (Vector3 (mousePos.x, mousePos.y, depth));

     transform.position = wantedPos;

}

File: Assets/Scripts/FollowMouse.cs
using UnityEngine;
using System.Collections;

public class FollowMouse : MonoBehaviour {

	public float depth;
	Vector3 mousePos, wantedPos;
	// Use this for initialization
	void Start () {
	
		Cursor.visible = false;
	}
	
	// Update is called once per frame
	void Update () {
	
		mousePos = Input.mousePosition;
		wantedPos = Camera.main.ScreenToWorldPoint (new Vector3 (mousePos.x, mousePos.y, depth));
		transform.position = wantedPos;
	}
}
